src/containers/Accounts/AMM/AMMAccounts/test/AMMAccounts.test.tsx

```diff
@@ -0,0 +1,124 @@
+import i18n from '../../../../../i18nTestConfig'
+import { initialState } from '../../../../../rootReducer'
+import * as rippled from '../../../../../rippled/lib/rippled'
+import NoMatch from '../../../../NoMatch'
+import { AMMAccountHeader } from '../AMMAccountHeader/AMMAccountHeader'
+import { AccountTransactionTable } from '../../../AccountTransactionTable'
+import { AMMAccounts } from '../index'
+
+const configureMockStore = require('redux-mock-store').default
+
+describe('AMM Account Page', () => {
+  const TEST_ACCOUNT_ID = 'rTEST_ACCOUNT'
+  const middlewares = [thunk]
+  const mockStore = configureMockStore(middlewares)
+  const accountTransactions: any = {
+    transactions: [
+      {
+        tx: {
+          Amount: '10000000000',
+          Amount2: { currency: 'USD', amount: '100000', issuer: 'SOLO' },
+        },
+      },
+    ],
+  }
+
+  const ammInfo: any = {
+    amm: {
+      Amount: '10000000000',
+      Amount2: { currency: 'USD', value: '100000' },
+      TradingFee: 10,
+      LPToken: {
+        value: '8989',
+      },
+    },
+  }
+
+  const createWrapper = (state = {}) => {
+    const store = mockStore({ ...initialState, ...state })
+    return mount(
+      <I18nextProvider i18n={i18n}>
+        <Provider store={store}>
+          <MemoryRouter initialEntries={[`accounts/${TEST_ACCOUNT_ID}`]}>
+            <Route path="accounts/:id" component={AMMAccounts} />
+          </MemoryRouter>
+        </Provider>
+      </I18nextProvider>,
+    )
+  }
+
+  it('renders AMM account page when TVL not present', async () => {
+    const state = {
+      ...initialState,
+    }
+
+    setSpy(accountTransactions, ammInfo)
+
+    const wrapper = createWrapper(state)
+    await flushPromises()
+    wrapper.update()
+    expect(wrapper.find(AMMAccountHeader).length).toBe(1)
+    expect(wrapper.find(AccountTransactionTable).length).toBe(1)
+    wrapper.unmount()
+  })
+
+  it('shows error when amm info data is formatted incorrectly', async () => {
+    const state = {
+      ...initialState,
+    }
+
+    setSpy(accountTransactions, 'ammInfo')
+
+    const wrapper = createWrapper(state)
+    await flushPromises()
+    wrapper.update()
+    expect(wrapper.find(AMMAccountHeader).length).toBe(0)
+    expect(wrapper.find(AccountTransactionTable).length).toBe(0)
+    expect(wrapper.find(NoMatch).length).toBe(1)
+    wrapper.unmount()
+  })
+
+  it('shows error when account transactions data is formatted incorrectly', async () => {
+    const state = {
+      ...initialState,
+    }
+    const accTransBad: any = {
+      transactions: [],
+    }
+
+    setSpy(accTransBad, ammInfo)
+
+    const wrapper = createWrapper(state)
+    await flushPromises()
+    wrapper.update()
+    expect(wrapper.find(AMMAccountHeader).length).toBe(0)
+    expect(wrapper.find(AccountTransactionTable).length).toBe(0)
+    expect(wrapper.find(NoMatch).length).toBe(1)
+    wrapper.unmount()
+  })
+})
+
+function flushPromises() {
```

Comment:
Doesn't this function need to be async?

Reply:
typescript infers it from the promise returned.